### PR TITLE
update pypdfium2 dep and improve code quality

### PR DESCRIPTION
**Pull request by felixdittrich92**
This PR:

- update pypdfium to 2.1.0 (to use with statement) and set fix to <3.0.0 to avoid #947 in upcoming releases
- improve code quality

Any feedback is welcome :hugs:

### CODE REVIEW

File: doctr/io/pdf.py [hunk 1]
```diff
@@ -41,8 +41,5 @@ def read_pdf(
         raise TypeError("unsupported object type for argument 'file'")
 
     # Rasterise pages to PIL images with pypdfium2 and convert to numpy ndarrays
-    pdf = pdfium.PdfDocument(file, password=password)
-    pages = [np.asarray(img) for img in pdf.render_topil(scale=scale, **kwargs)]
-    pdf.close()
-
-    return pages
+    with pdfium.PdfDocument(file, password=password) as pdf:
+        return [np.asarray(img) for img in pdf.render_topil(scale=scale, **kwargs)]
```
Comment (mara004) [inline doctr/io/pdf.py:45]: I'm not certain what impact a `return` statement inside the context manager might have. Will this actually call `__close__()`? To be safe, I'd rather cleanly assign  `pages = [...]` and then put the return outside the context manager.

Comment (mara004) [reply doctr/io/pdf.py:45]: Ah, according to [this thread](https://stackoverflow.com/questions/9885217/in-python-if-i-return-inside-a-with-block-will-the-file-still-close), it's safe to do this. You never stop learning!

Comment (felixdittrich92) [reply doctr/io/pdf.py:45]: Should be fine close is called like in a finally block so after returning :)

**Comment by mara004** [comment]
Looks good, then!

**Review by frgfm — APPROVED** [review]
Thanks Felix, I was about to open a PR for this specifically :)
We should update the version specifiers of other dependencies accordingly I think

**Comment by felixdittrich92** [comment]
> Thanks Felix, I was about to open a PR for this specifically :)
> We should update the version specifiers of other dependencies accordingly I think

:sweat_smile:  

About the deps part let us check if we can handle this with depend-a-bot this would make it much easier (really) ^^

**Comment by frgfm** [comment]
That is a different problem though: dependabot is (to the best of my knowledge) reporting version specifier updates for security reasons

Here we need to reassess the upper and lower bound of each dep: I'll take care of it, and then set the dependabot :+1:

**Pull request merged**